### PR TITLE
Deprecated StatsD Exporter (fixes #82)

### LINKED ISSUE #82: Deprecate maddox/statsd-exporter
The Envoy image has been updated to v1.11 that has native Prometheus support for HTTP and gPRC metrics including latency `envoy_cluster_upstream_rq_time_bucket`. I think that the statsd-exporter sidecar is no longer needed.

@PaulMaddox do you agree on removing it or are there any metrics that are missing from Envoy native prom exporter?

### PR DESCRIPTION
**Pull request by PaulMaddox**
*Issue #, if available:* #82 

*Description of changes:* Previously the envoy prometheus exporter did not emit metrics around request latencies, and it was necessary to include a sidecar for exporting statsd metrics to obtain these. This is no longer the case, so the statsd exporter sidecar is not required any more. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,16 +25,11 @@ $ export MESH_NAME=my_mesh_name
 $ export ENABLE_STATS_TAGS=true
 ```
 
-(Optional) If enabled, Envoy will emit DogStatsD metrics to 127.0.0.1:8125, where it expects to find a statsd receiver. This could be either a Datadog sidecar, or something like [statsd_exporter](https://github.com/prometheus/statsd_exporter) (see below).
+(Optional) If enabled, Envoy will emit DogStatsD metrics to 127.0.0.1:8125, where it expects to find a statsd receiver.
```
Comment (nckturner) [inline README.md]: So I looked at the docs, and the appmesh envoy accepts an `ENABLE_ENVOY_DOG_STATSD` env var.   Is that configurable by the injector after this change?

File: cmd/app-mesh-inject/main.go [hunk 2]
```diff
@@ -65,7 +65,6 @@ func init() {
 	flag.BoolVar(&cfg.InjectXraySidecar, "inject-xray-sidecar", false, "Enable Envoy X-Ray tracing integration and injects xray-daemon as sidecar")
 	flag.BoolVar(&cfg.EnableStatsTags, "enable-stats-tags", false, "Enable Envoy to tag stats")
 	flag.BoolVar(&cfg.EnableStatsD, "enable-statsd", false, "If enabled, Envoy will send DogStatsD metrics to 127.0.0.1:8125")
-	flag.BoolVar(&cfg.InjectStatsDExporterSidecar, "inject-statsd-exporter-sidecar", false, "Deploy statsd_exporter as a sidecar listening on 127.0.0.1:8125 to recieve metrics and export them in prometheus format on 127.0.0.1:9102")
```
Comment (nckturner) [inline cmd/app-mesh-inject/main.go]: I think we should leave the flag and just print out a deprecation message, that it doesn't do anything.  That we we don't break anyone for at least a release.

Comment (PaulMaddox) [reply cmd/app-mesh-inject/main.go]: Agreed. Have updated the PR.

File: README.md (hunk 1 above)
Comment (PaulMaddox) [reply README.md]: Yes. The `--enable-statsd` flag (which isn't removed by this PR) defines whether or not `ENABLE_ENVOY_DOG_STATSD` is set on the envoy container. See https://github.com/aws/aws-app-mesh-inject/blob/c8ac4885b663afe1fd6a8e31c73cdbcd130dc1b1/pkg/patch/sidecar.go#L51-L55

**Comment by PaulMaddox** [comment]
Is there something up with the TravisCI job? It seems to have run on my initial PR, but the new commits I've added don't seem to have triggered it. 

I have manually tested the new deprecation warning commit by building on my machine (via the Dockerfile), and testing that the deprecation warning shows successfully if the flag is used.

**Pull request merged**